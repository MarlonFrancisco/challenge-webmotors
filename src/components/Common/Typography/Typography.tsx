import React, { CSSProperties } from "react";
import { useTheme } from "styled-components";

type TypographyProps = {
  variant?: "p" | "span" | "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
  style?: CSSProperties;
  color?: "primary" | "secondary";
  size?: number;
  weight?: number;
  align?: "center" | "right" | "left";
};

const Typography: React.FC<TypographyProps> = ({
  children,
  variant = "h1",
  color = "primary",
  size = 16,
  weight = 500,
  style = {},
  align = "center",
}) => {
  const theme = useTheme();
  return React.createElement(
    variant,
    {
      style: {
        fontSize: `${size}px`,
        color: theme.palette.text[color],
        fontWeight: weight,
        textAlign: align,
        ...style,
      },
    },
    children
  );
};

export default Typography;
