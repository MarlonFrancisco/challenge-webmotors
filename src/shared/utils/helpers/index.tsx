import React from "react";
import { ThemeProvider } from "styled-components";
import { theme } from "../../styled";

type WithThemeProps = {};

export const WithTheme: React.FC<WithThemeProps> = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);
