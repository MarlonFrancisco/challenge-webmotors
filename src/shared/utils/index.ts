export type Options = {
  withoutPrefix: boolean;
  splitted: boolean;
};

export const formatCurrency = (
  value = 0,
  options?: Options
): string | string[] => {
  const formatted = new Intl.NumberFormat("pt-BR", {
    style: "currency",
    currency: "BRL",
  }).format(value);
  let apllyOptions: string | string[] = formatted;

  if (!options) {
    return formatted;
  }

  if (options.withoutPrefix) {
    apllyOptions = apllyOptions.replace("R$", "").trim();
  }

  if (options.splitted) {
    apllyOptions = apllyOptions.split(",");
  }

  return apllyOptions;
};
