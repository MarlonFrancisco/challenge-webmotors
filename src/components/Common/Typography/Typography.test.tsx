import React from "react";
import { render } from "@testing-library/react";
import Typography, { TypographyProps } from "./Typography";
import { WithTheme } from "../../../shared/utils/helpers";

const Wrapper = (props: TypographyProps) => (
  <WithTheme>
    <Typography {...props} />
  </WithTheme>
);

describe("Typography", () => {
  let props: TypographyProps & { children: string };

  beforeEach(() => {
    props = {
      children: "webmotors",
    };
  });

  it("should render without an error", () => {
    const { container } = render(<Wrapper {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("should render props", () => {
    const { queryByText } = render(<Wrapper {...props} />);

    expect(queryByText(props.children)).not.toBeNull();
  });
});
