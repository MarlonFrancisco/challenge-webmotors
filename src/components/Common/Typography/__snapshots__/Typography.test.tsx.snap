// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should render without an error 1`] = `
<div>
  <h1
    style="font-size: 16px; color: rgb(186, 186, 188); font-weight: 500; text-align: center;"
  >
    webmotors
  </h1>
</div>
`;
